File: src/chapter3/code3.2.ts
function code3_2(){
    //線形探索法
    //条件を満たすものが存在する場所も知る
    //aの中からNを探す
    const a = [4,3,12,7,11]
    const N = 7

    var found_id = -1 //ありえない値で初期化しておく
    for (let i = 0; i < a.length; i++) {
        if(a[i] === N){
            found_id = i //見つけた添字を記録
            break
        }
    }
    
    console.log(found_id)
}
code3_2()

File: src/chapter3/code3.2.cpp
#include <iostream>
#include <vector>
using namespace std;

//特定の要素の存在する添字(インデックス)を取得する
int main(){
    //入力を受け取る
    int N, v;
    cin >> N >> v;
    vector<int> a(N);
    for(int i = 0; i < N; ++i) cin >> a[i];

    //線形探索
    int found_id = -1; //インデックスとしてありえない数値で初期化
    for (int i = 0; i < N; i++){
        if(a[i] == v){
            found_id = i; //見つけた添字を記録
            break;
        }
    }
    
    //結果出力(-1のときは見つかっていない)
    cout << found_id << endl;
}